### PR TITLE
Setup a proper CPE pipeline.

### PR DESCRIPTION
**Pull request by felicianotech**
This PR is to get the Go orb up-to-snuff with how the CPE Team will be properly running CI pipelines for our orbs. This setup allows integration testing.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,39 +1,91 @@
...
+      - integration-test-2
+
+      # publish a semver version of the orb. relies on
+      # the commit subject containing the text "[semver:patch|minor|major|skip]"
+      # as that will determine whether a patch, minor or major
+      # version will be published or if publishing should
+      # be skipped.
+      # e.g. [semver:patch] will cause a patch version to be published.
+      - orb-tools/dev-promote-prod-from-commit-subject:
+          orb-name: circleci/aws-s3
```
Comment (KyleTryon) [inline .circleci/config.yml]: Orb name must be changed.

Comment (felicianotech) [reply .circleci/config.yml]: Nice catch. Fixed.

Comment (KyleTryon) [inline .circleci/config.yml]: Orb name must be changed.

**Comment by felicianotech** [comment]
@KyleTryon  Your comment showed up twice somehow but it has been fixed.

**Comment by felicianotech** [comment]
Why skip?

**Comment by KyleTryon** [comment]
> Why skip?

My thinking was no new functionality was added to the orb itself.

**Pull request merged**